apps/condo/domains/common/components/UseDeskWidget.tsx

```diff
@@ -1,60 +1,80 @@
     }
 
-    return get(window, ['usedeskMessenger', 'userIdentify'], null)
+    return get(window, 'usedeskMessenger', null)
 }
 
 const UseDeskWidget: React.FC = () => {
     const { link } = useOrganization()
     const { user } = useAuth()
 
-    const userIdentify = getUserIdentify()
+    const messenger = useMemo(() => getUsedeskMessenger(), [])
+    const userIdentify = useMemo(() => get(messenger, 'userIdentify', null), [messenger])
 
     useEffect(() => {
         try {
-            if (UseDeskWidgetId && isFunction(userIdentify)) {
+            if (UseDeskWidgetId && isFunction(userIdentify) && typeof window !== 'undefined') {
                 const name = get(link, 'name')
                 const email = get(user, 'email')
                 const phone = get(user, 'phone')
-                userIdentify({
-                    name,
-                    email,
-                    phone: typeof phone === 'string' ? phone.slice(1) : phone,
-                    additional_fields:
-                        [
-                            {
-                                id: useDeskFieldsIdsMap.tin, value: get(link, ['organization', 'tin'], null),
-                            },
-                            {
-                                id: useDeskFieldsIdsMap.organizationName, value: get(link, ['organization', 'name'], null),
-                            },
-                            {
-                                id: useDeskFieldsIdsMap.role, value: get(link, ['role', 'name'], null),
-                            },
-                        ],
+
+                set(window, '__widgetInitCallback', (widget) => {
+                    const token = cookie.get('usedeskChatToken')
+
+                    const baseData = {
+                        name,
+                        email,
+                        phone: typeof phone === 'string' ? phone.slice(1) : phone,
+                        additional_fields:
+                            [
+                                {
+                                    id: useDeskFieldsIdsMap.tin, value: get(link, ['organization', 'tin'], null),
+                                },
+                                {
+                                    id: useDeskFieldsIdsMap.organizationName, value: get(link, ['organization', 'name'], null),
+                                },
+                                {
+                                    id: useDeskFieldsIdsMap.role, value: get(link, ['role', 'name'], null),
+                                },
+                            ],
+                    }
+
+                    widget.userIdentify(token ? { ...baseData, token } : baseData)
+                })
+
+                set(window, '__usedeskWidgetFirstClientMessageCallback', () => {
+                    const token = messenger.getChatToken()
+                    const currentDate = new Date()
+                    currentDate.setMinutes(currentDate.getMinutes() + 5)
```

Comment:
What this tweak of minutes is about?

Reply:
For testing, the cookie’s lifetime is not 2 days, but 5 minutes.
Test passed, returned 2 days